### ISSUE TITLE
Incorrect detection for DS001 for multi-stage builds

### ISSUE THREAD

**Issue by MShekow** [aquasecurity/appshield#129]
Trivy incorrectly detects that I'm using a ":latest" tag. Reproducible with a trivial file such as this:

```
FROM node:14.18.1-bullseye AS dependencies

FROM dependencies AS build
```

**Comment by AErmie** [comment]
I'm encountering the same issue, and we don't want to add an exception in the .trivyignore file since it will prevent it from catching legitimate scenarios.

**Comment by MShekow** [comment]
How can I verify whether the fix works? The example I posted above still exhibits DS001 for Trivy 0.21.3 (which was released after you merged this fix).

**Comment by knqyf263** [comment]
We have released AppShield v1.0.1 just now. You can try it with Trivy since Trivy downloads the latest policy.
